molgenis-data-import/src/main/java/org/molgenis/data/importer/ImportedDataToBackendCompatibilityValidator.java

```diff
@@ -0,0 +1,45 @@
+package org.molgenis.data.importer;
+
+import org.molgenis.data.DataService;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.MolgenisDataException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.google.common.collect.ImmutableCollection;
+
+/**
+ * This class serves as an early validator. It can check if an entity is valid before we process parsed data into
+ * repositories.
+ *
+ */
+public class ImportedDataToBackendCompatibilityValidator
+{
+	@Autowired
+	DataService dataService;
+
+	public ImportedDataToBackendCompatibilityValidator(DataService dataService)
+	{
+		this.dataService = dataService;
+	}
+
+	public void validate(ImmutableCollection<EntityMetaData> immutableCollection)
+	{
+		immutableCollection.forEach(entity -> checkIfEntityExists(entity));
+	}
+
+	private void checkIfEntityExists(EntityMetaData entity)
+	{
+		if (dataService.hasRepository(entity.getName()))
```

Comment:
I think the dataservice holds all repositories, not only those you have rights to?